Ronin.rte/Devices/Weapons/RPG7/RPG7Pie.lua

```diff
@@ -0,0 +1,12 @@
+function RoninRPGSwitchAmmo(actor)
+	local gun = ToAHuman(actor).EquippedItem;
+	if gun and actor:HasObject("Shovel") then
+		local gun = ToHDFirearm(gun);
+		local magSwitchName = "Magazine Ronin Shovel Shot";
+		if gun.Magazine == nil or gun.Magazine.PresetName ~= magSwitchName then
+			actor:RemoveInventoryItem("Shovel");
+			gun:SetNextMagazineName(magSwitchName);
+			gun:Reload();
```

Comment:
Can you not do gun:SetNextMagazineName("Magazine Ronin RPG-7") here and have it work? Would be nice to get rid of the need for an Update function on RPG7, or at least disable it when it's not relevant (i.e. enable it in the ammo switch function then disable it after it's run once).

Also, probably not helpful in this case but don't forget that you can set Magazine directly with 143 all merged in :)

Reply:
What? No..? You can only specify one "next" magazine at a time!

I suppose I could disable the script when it's not needed, but wouldn't that also disable the consequent OnPieMenu script?

Reply:
Sorry, that was unclearly written - if you do
```
gun:SetNextMagazineName(magSwitchName);
gun:Reload();
gun:SetNextMagazineName("Magazine Ronin RPG-7");
```
will it not work?

As for disabling script stuff, you'd have to separate it into 2 script files, yeah.

Reply:
Yeah no, doesn't work that way - the next magazine name can be overwritten all the way until the actual magazine is added.